app/src/androidTest/java/protect/card_locker/LanguageChangeUITest.java

```diff
@@ -0,0 +1,178 @@
+        textView2.check(matches(withText("Ρυθμίσεις")));
+
+        ViewInteraction recyclerView3 = onView(
+                allOf(withId(com.jaredrummler.android.colorpicker.R.id.recycler_view),
+                        childAtPosition(
+                                withId(android.R.id.list_container),
+                                0)));
+        recyclerView3.perform(actionOnItemAtPosition(4, click()));
+
+        DataInteraction appCompatCheckedTextView3 = onData(anything())
+                .inAdapterView(allOf(withId(androidx.appcompat.R.id.select_dialog_listview),
+                        childAtPosition(
+                                withId(androidx.appcompat.R.id.contentPanel),
+                                0)))
+                .atPosition(0);
+        appCompatCheckedTextView3.perform(click());
+
+        ViewInteraction appCompatImageButton = onView(
+                allOf(withContentDescription("Navigate up"),
+                        childAtPosition(
+                                allOf(withId(R.id.toolbar),
+                                        childAtPosition(
+                                                withClassName(is("com.google.android.material.appbar.AppBarLayout")),
+                                                0)),
+                                1),
+                        isDisplayed()));
+        appCompatImageButton.perform(click());
+
+        ViewInteraction textView3 = onView(
+                allOf(withId(R.id.welcome_text), withText("Welcome to Catima"),
+                        withParent(allOf(withId(R.id.helpSection),
+                                withParent(withId(R.id.include)))),
+                        isDisplayed()));
+        textView3.check(matches(withText("Welcome to Catima")));
+    }
+
+    private static Matcher<View> childAtPosition(
+            final Matcher<View> parentMatcher, final int position) {
+
+        return new TypeSafeMatcher<View>() {
+            @Override
+            public void describeTo(Description description) {
+                description.appendText("Child at position " + position + " in parent ");
+                parentMatcher.describeTo(description);
+            }
+
+            @Override
+            public boolean matchesSafely(View view) {
+                ViewParent parent = view.getParent();
+                return parent instanceof ViewGroup && parentMatcher.matches(parent)
+                        && view.equals(((ViewGroup) parent).getChildAt(position));
+            }
+        };
+    }
+
+    private static int getPositionOfItem(String selectedLanguage) {
+        String[] availableLocales = getInstrumentation().getTargetContext().getResources().getStringArray(R.array.locale_values);
+        int position = 0;
+        for (String locale : availableLocales) {
+            if (locale.equals(selectedLanguage)) {
+                break;
+            }
+            position++;
+        }
+        return position;
```

Comment:
```suggestion
        return Arrays.asList(availableLocales).indexOf(selectedLanguage);
```

Reply:
Thank you for your suggestion! I've integrated your changes into the PR.